handlers/secrets.go

```diff
@@ -28,30 +29,70 @@ const (
 
 // MakeSecretHandler makes a handler for Create/List/Delete/Update of
 //secrets in the Kubernetes API
-func MakeSecretHandler(namespace string, kube kubernetes.Interface) http.HandlerFunc {
+func MakeSecretHandler(defaultNamespace string, kube kubernetes.Interface) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		if r.Body != nil {
 			defer r.Body.Close()
 		}
 
-		client := kube.CoreV1().Secrets(namespace)
+		lookupNamespace, err, errCode := getLookupNamespace(defaultNamespace, r)
+		if err != nil {
+			http.Error(w, err.Error(), errCode)
+		}
+
+		client := kube.CoreV1().Secrets(lookupNamespace)
 
 		switch r.Method {
 		case http.MethodGet:
 			getSecretsHandler(client, w, r)
 		case http.MethodPost:
-			createSecretHandler(client, namespace, w, r)
+			createSecretHandler(client, lookupNamespace, w, r)
 		case http.MethodPut:
-			replaceSecretHandler(client, namespace, w, r)
+			replaceSecretHandler(client, lookupNamespace, w, r)
 		case http.MethodDelete:
-			deleteSecretHandler(client, namespace, w, r)
+			deleteSecretHandler(client, lookupNamespace, w, r)
 		default:
 			w.WriteHeader(http.StatusBadRequest)
 			return
 		}
 	}
 }
 
+// getLookupNamespace returns the namespace specified in any request type
+func getLookupNamespace(defaultNamespace string, r *http.Request) (string, error, int) {
+	lookupNamespace := defaultNamespace
+
+	if r.Method == http.MethodGet {
+		q := r.URL.Query()
+		namespace := q.Get("namespace")
+
+		if len(namespace) > 0 {
+			lookupNamespace = namespace
+		}
+	} else {
+		body, _ := ioutil.ReadAll(r.Body)
+		req := types.Secret{}
```

Comment:
you could also use an anonymous struct here, which will possibly make it more flexible

```go
req := struct{ Namespace string }{}
```

This method could then be used for any POST/PUT/PATCH request that has a `Namespace` field in the body.

In that case, you could use a switch statement on `r.Method`  to handle all of these cases

```go
switch r.Method {
case http.MethodPost, http.MethodPut, http.Patch:
    // load req body
default:
    // load from GET param
}
```

Reply:
I like this, will implement!

Reply:
I like this, will implement!